pydough/conversion/hybrid_translator.py

```diff
@@ -760,6 +768,93 @@ def rewrite_median_call(
 
         return avg_call
 
+    def rewrite_quantile_call(
+        self,
+        child_connection: HybridConnection,
+        expr: HybridFunctionExpr,
+        create_new_calc: bool,
+    ) -> HybridFunctionExpr:
+        """
+        Rewrites a QUANTILE aggregation call into an equivalent expression using window functions.
+        This is typically used for dialects that do not natively support the PERCENTILE_DISC
+        aggregate function.
+
+        The rewritten expression selects the value at the specified quantile by:
+        - Ranking the rows within each partition.
+        - Calculating the number of rows (N) in each partition.
+        - Keeping only those rows where the rank is greater than INTEGER((1.0 - p) * N),
+            where p is the quantile argument.
+        - Taking the maximum value among the kept rows.
+
+        Args:
+            child_connection: The HybridConnection containing the aggregate call to QUANTILE.
+            expr: The HybridFunctionExpr representing the QUANTILE aggregation.
+            create_new_calc: If True, injects new expressions into a new CALCULATE operation.
+
+        Returns:
+            A HybridFunctionExpr representing the rewritten aggregation using window functions.
+        """
+        assert expr.operator == pydop.QUANTILE
+
+        # Valid if the value of p is a number between 0 and 1
+        if (
+            not isinstance(expr.args[1], HybridLiteralExpr)
+            or not isinstance(expr.args[1].typ, NumericType)
+            or not isinstance(expr.args[1].literal.value, (int, float))
+            or not (0.0 <= float(expr.args[1].literal.value) <= 1.0)
+        ):
+            raise ValueError(
+                f"Expected second argument to QUANTILE to be a numeric literal between 0 and 1, instead received {expr.args[1]!r}"
+            )
+
+        assert len(expr.args) == 2
+        # The implementation
+        # MAX(KEEP_IF(args[0], R > INTEGER((1.0-args[1]) * N)))
+        data_expr: HybridExpr = expr.args[0]  # Column
+        p: HybridExpr = expr.args[1]  # percentage
+        one: HybridExpr = HybridLiteralExpr(Literal(1.0, NumericType()))
+
+        assert child_connection.subtree.agg_keys is not None
+        partition_args: list[HybridExpr] = child_connection.subtree.agg_keys
+        order_args: list[HybridCollation] = [HybridCollation(data_expr, False, False)]
+
+        # R
+        rank: HybridExpr = HybridWindowExpr(
+            pydop.RANKING, [], partition_args, order_args, NumericType(), {}
+        )
+        # N
+        rows: HybridExpr = HybridWindowExpr(
+            pydop.RELCOUNT, [data_expr], partition_args, [], NumericType(), {}
+        )
+
+        # (1.0-args[1])
+        sub: HybridExpr = HybridFunctionExpr(pydop.SUB, [one, p], NumericType())
```

Comment:
Since we know `p` is a numeric literal, we can forgo this expr entirely and just pass in the new literal:
```suggestion
        sub: HybridExpr = HybridLiteralExpr(Literal(1.0 - p.literal.value, NumericType()))
```

Reply:
Changed. It changed the sql, just make sure everything still looks fine